source/common/quic/quic_server_transport_socket_factory.cc

```diff
@@ -26,11 +27,77 @@ QuicServerTransportSocketConfigFactory::createTransportSocketFactory(
 
   auto factory = std::make_unique<QuicServerTransportSocketFactory>(
       PROTOBUF_GET_WRAPPED_OR_DEFAULT(quic_transport, enable_early_data, true),
-      context.statsScope(), std::move(server_config));
+      context.statsScope(), std::move(server_config), context.sslContextManager(), server_names);
   factory->initialize();
   return factory;
 }
 
+namespace {
+void initializeQuicCertAndKey(Ssl::TlsContext& context,
+                              const Ssl::TlsCertificateConfig& /*cert_config*/) {
+  // Convert the certificate chain loaded into the context into PEM, as that is what the QUICHE
+  // API expects. By using the version already loaded, instead of loading it from the source,
+  // we can reuse all the code that loads from different formats, allows using passwords on the key,
+  // etc.
+  std::string certs_str;
+  auto process_one_cert = [&](X509* cert) {
```

Comment:
nit: Would it make sense to just make this function in the anonymous namespace?

Reply:
I prefer having it in the function that uses it to make it clear the scope in which it's used, and it can capture a variable instead of needing a parameter. But it's just a small style preference. I'll change it if you insist.